jwst/cube_build/cube_cloud.py

```diff
@@ -60,53 +62,70 @@ def match_det2cube_msm(naxis1, naxis2, naxis3,
         # cube coordinates.
         # find the spaxels that fall withing ROI of point cloud defined  by
         # coord1,coord2,wave
-
+        lower_limit = softrad_pixel[ipt]
         xdistance = (xcenters - coord1[ipt])
         ydistance = (ycenters - coord2[ipt])
         radius = np.sqrt(xdistance * xdistance + ydistance * ydistance)
-        indexr = np.where(radius <= rois)
-        indexz = np.where(abs(zcoord - wave[ipt]) <= roiw)
-
-        d1 = np.array(coord1[ipt] - xcenters[indexr]) / cdelt1
-        d2 = np.array(coord2[ipt] - ycenters[indexr]) / cdelt2
-        d3 = np.array(wave[ipt] - zcoord[indexz]) / cdelt3
-        dxy = (d1 * d1) + (d2 * d2)
+        indexr = np.where(radius <= rois_pixel[ipt])
+        indexz = np.where(abs(zcoord - wave[ipt]) <= roiw_pixel[ipt])
+
+        # on the wavelength boundaries the point cloud may not be in the IFUCube
+        # the edge cases are skipped and not included in final IFUcube.
+        # Left commented code for checking later for NIRSPEC the spectral size
+        # in the reference file may be too small
+#        if len(indexz[0]) == 0:
+#            if wave[ipt] < zcoord[0]:
+#               ilow = ilow + 1
+#            elif wave[ipt] > zcoord[-1]: 
+#               ihigh = ihigh + 1
+#            else:
+#                imatch = imatch + 1
+#                print(' no z match found ',wave[ipt],roiw_pixel[ipt])
+#                print(zcoord[naxis3-11:naxis3])
+#                diff = abs(zcoord[naxis3-11:naxis3] - wave[ipt])
+#                print(diff)
+#                exit()
```

Comment:
Remove these commented-out lines (and the ones below).

Reply:
The reason I left them in is because I am waiting for a new and improved MIRI cubepars reference file. The current reference file for long wavelengths is too fine in the wavelength dimension or the roi in the spectral dimension is too small. There are detector values that do not match with any cube spaxel. Ben Sargent makes the cubepars reference files and is waiting for an improved MIRI simulator to make work out what the values in the reference file should be. 
I left this code in because I can check to make sure the new reference file is working better. 
Once we have this sorted out I can see if I want to make this a debug option.

I did not want to do that at this time - but I could if you prefer ?